umh-core/pkg/service/s6/lifecycle_force.go

```diff
@@ -0,0 +1,151 @@
+// Copyright 2025 UMH Systems GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package s6
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/united-manufacturing-hub/united-manufacturing-hub/umh-core/pkg/service/filesystem"
+)
+
+const (
+	// gracePeriodForTermination is the grace period to wait between SIGTERM and SIGKILL
+	gracePeriodForTermination = 500 * time.Millisecond
+)
+
+// ForceCleanup performs aggressive cleanup for stuck services
+// Uses expert-recommended patterns:
+// - Process termination and supervisor killing
+// - Comprehensive artifact removal
+func (s *DefaultService) ForceCleanup(ctx context.Context, artifacts *ServiceArtifacts, fsService filesystem.Service) error {
+	if s == nil {
+		return fmt.Errorf("lifecycle manager is nil")
+	}
+
+	if ctx.Err() != nil {
+		return ctx.Err()
+	}
+
+	if artifacts == nil {
+		return fmt.Errorf("artifacts is nil")
+	}
+
+	s.logger.Warnf("Force cleaning service artifacts: service=%s, files=%d",
+		filepath.Base(artifacts.ServiceDir), len(artifacts.CreatedFiles))
+
+	// Create down files first
+	if err := s.createDownFiles(ctx, artifacts, fsService); err != nil {
+		s.logger.Warnf("Failed to create down files: %v", err)
+		// Continue with cleanup even if down files fail
+	}
+
+	// Best-effort process termination
+	if err := s.terminateProcesses(ctx, artifacts, fsService); err != nil {
```

Comment:
What happens if only a supervisor remains.
Will this early abort or continue to the supervisor killing ?